src/farmer.rs

```diff
@@ -36,34 +40,75 @@ impl PlotDescription {
 
 #[derive(Default)]
 pub struct Builder {
-    listen_on: Option<Multiaddr>,
-    ws_rpc: Option<SocketAddr>,
+    _ensure_cant_construct: (),
+    // TODO: add once those will be used
+    // listen_on: Option<Multiaddr>,
+    // ws_rpc: Option<SocketAddr>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum BuildError {
+    #[error("Single disk plot creation error: {0}")]
+    SingleDiskPlotCreate(#[from] SingleDiskPlotError),
+    #[error("No plots error")]
+    NoPlotsSupplied,
 }
 
 impl Builder {
     pub fn new() -> Self {
         Self::default()
     }
 
-    pub fn listen_on(mut self, multiaddr: Multiaddr) -> Self {
-        self.listen_on = Some(multiaddr);
-        self
-    }
-
-    pub fn ws_rpc(mut self, ws_rpc: SocketAddr) -> Self {
-        self.ws_rpc = Some(ws_rpc);
-        self
-    }
+    // TODO: add once those will be used
+    // pub fn listen_on(mut self, multiaddr: Multiaddr) -> Self {
+    //     self.listen_on = Some(multiaddr);
+    //     self
+    // }
+    //
+    // pub fn ws_rpc(mut self, ws_rpc: SocketAddr) -> Self {
+    //     self.ws_rpc = Some(ws_rpc);
+    //     self
+    // }
 
     /// It supposed to open node at the supplied location
     pub async fn build(
         self,
         reward_address: PublicKey,
         node: Node,
-        plots: Vec<PlotDescription>,
-    ) -> Result<Farmer, ()> {
-        let _ = (reward_address, node, plots);
-        todo!()
+        plots: &[PlotDescription],
+    ) -> Result<Farmer, BuildError> {
+        if plots.is_empty() {
+            return Err(BuildError::NoPlotsSupplied);
+        }
+
+        let mut single_disk_plots_stream = plots
```

Comment:
rust anaylzer on vscode got angry here for me:
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/32795992/197524227-7d575a08-bd82-43de-b4fe-aa154f1fc82d.png">